packages/connections/src/stores/connections-store.ts

```diff
@@ -351,6 +360,42 @@ export function useConnections(
           type: 'hide-store-connection-error',
         });
       },
+      async saveConnection(connectionInfo: ConnectionInfo) {
+        await saveConnectionInfo(connectionInfo);
+
+        const existingConnectionIndex = connections.findIndex(
+          (connection) => connection.id === connectionInfo.id
+        );
+
+        const newConnections = [...connections];
+
+        if (existingConnectionIndex !== -1) {
+          // Update the existing saved connection.
+          newConnections[existingConnectionIndex] = {
+            ...cloneDeep(newConnections[existingConnectionIndex]),
+            ...cloneDeep(connectionInfo),
```

Comment:
this may be prone to bugs, we may want to be more intentional here and just merge `lastUsed` .. although are we sure we need this? sorry for being extra pedantic about it, but this feels like something that may have unexpected side effects in the long run

Reply:
Good call, remove the cloning the previous.